### PR TITLE
Add Configurator context manager 'route_prefix_context' to allow for …

### PR DESCRIPTION
**Pull request by huntcsg**
…adding routes and including configuration callables with a particular route prefix.

### CODE REVIEW

**Comment by huntcsg** [comment]
#3218

**Comment by mmerickel** [comment]
Because this api mutates the internal state of the configurator I don't like having a return value for the context. I'd rather the code examples just be:

```python
with config.route_prefix_context('/foo/bar'):
    config.add_route(...)
```

File: pyramid/config/routes.py [hunk 1]
```diff
@@ -467,3 +468,49 @@ def get_routes_mapper(self):
...
+
+        >>> config = Configurator()
+        >>> with config.route_prefix_context('foo') as api_config:
+        ...     api_config.add_route('bar', '/bar')
+
+        Arguments
+
+        route_prefix
+
+         A string suitable to be used as a route prefix, or ``None``."""
```
[the remark this answers is not in the document]
Comment (mmerickel) [reply pyramid/config/routes.py]: This feature isn't going to be in 1.9, but 1.10 would be appropriate.

**Comment by huntcsg** [comment]
@stevepiercy I made the changes as you suggested.

**Comment by stevepiercy** [comment]
@mmerickel is this OK to merge, or do you require the change you prefer for the [usage example](https://github.com/Pylons/pyramid/pull/3279/files#diff-ff1bd60eed51d4ecb4f69704d77a6dcaR481)?

**Comment by mmerickel** [comment]
Thanks @huntcsg !

**Pull request merged**